pyproject.toml

```diff
@@ -5,11 +5,70 @@ requires = [
   "setuptools_scm[toml]>=7.0",
   "wheel",
   "oldest-supported-numpy",
-  "Cython>=3.0"
+  "Cython>=3.0",
 ]
 # Defined by PEP 517
 build-backend = "setuptools.build_meta"
 
+[project]
+name = "cf-units"
+authors = [
+    {name = "cf-units Contributors", email = "scitools.pub@gmail.com"}
+]
+classifiers = [
+    "Development Status :: 5 - Production/Stable",
+    "Intended Audience :: Science/Research",
+    "License :: OSI Approved :: BSD License",
+    "Operating System :: OS Independent",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3 :: Only",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3.12",
+    "Programming Language :: Python :: Implementation :: CPython",
+    "Topic :: Scientific/Engineering",
+    "Topic :: Scientific/Engineering :: Atmospheric Science",
+    "Topic :: Scientific/Engineering :: Oceanography",
+]
+dependencies = [
+    "antlr4-python3-runtime ==4.11.1",   # To update this, see cf_units/_udunits2_parser/README.md
+    "cftime >=1.2",
+    "jinja2",
+    "numpy ==1.26.4"
+    # udunits2 cannot be installed with pip, and it is expected to be
+    #  installed separately.
+]
+description = "Units of measure as required by the Climate and Forecast (CF) metadata conventions"
+dynamic = [
+    "version",
+]
+keywords = [
+    "units",
+    "cf",
+    "cf-metadata",
+    "netcdf",
+    "science",
+    "earth-science",
+    "oceanography",
+    "meteorology",
+    "climate",
+    "python",
+]
+license.file = "LICENSE"
+readme = "README.md"
+requires-python = ">=3.9"
+
+[project.optional-dependencies]
+docs = ["sphinx"]
+test = ["codecov", "cython", "jinja2", "pip", "pytest", "pytest-cov"]
+all = ["pre-commit", "sphinx", "codecov", "cython", "jinja2", "pip", "pytest", "pytest-cov"]
```

Comment:
This one is a bit questionable to me. `cf-units[docs,test]` is sufficient, if needed at all :wink:

Reply:
I was lead to believe that TOML does not support any dynamic content. Or are you implying that the only thing/person reading this is a human?

Reply:
Oh wait you mean removing `all = ` entirely, don't you?

Reply:
That's ace! 9cf0ace63d4532cfdaa555de3097fbd315367af7

Reply:
I think this **_is_** now resolved.
https://github.com/SciTools/cf-units/commit/e29330832a78e87b0a8bef5558fb889b8e8a6b27#diff-50c86b7ed8ac2cf95bd48334961bf0530cdc77b5a56f852c5c61b89d735fd711R61-R64